src/components/customers/AddWalletToCustomerDialog.tsx

```diff
@@ -0,0 +1,220 @@
+} from '~/generated/graphql'
+import { addToast } from '~/core/apolloClient'
+
+gql`
+  mutation createCustomerWallet($input: CreateCustomerWalletInput!) {
+    createCustomerWallet(input: $input) {
+      id
+    }
+  }
+`
+
+export interface AddWalletToCustomerDialogRef extends DialogRef {}
+
+interface AddWalletToCustomerDialogProps {
+  customerId: string
+}
+
+export const AddWalletToCustomerDialog = forwardRef<DialogRef, AddWalletToCustomerDialogProps>(
+  ({ customerId }: AddWalletToCustomerDialogProps, ref) => {
+    const { translate } = useInternationalization()
+    const [createWallet] = useCreateCustomerWalletMutation({
+      onCompleted(res) {
+        if (res?.createCustomerWallet) {
+          addToast({
+            severity: 'success',
+            translateKey: 'text_62d6d5739e4eee96c1afaee8',
+          })
+        }
+      },
+    })
+
+    const formikProps = useFormik<Omit<CreateCustomerWalletInput, 'customerId'>>({
+      initialValues: {
+        expirationDate: undefined,
+        grantedCredits: '',
+        name: '',
+        paidCredits: '',
+        rateAmount: '1.00',
+      },
+      validationSchema: object().shape({
+        expirationDate: date(),
+        grantedCredits: string().required(''),
+        name: string(),
+        paidCredits: string().required(''),
+        rateAmount: string().required(''),
+      }),
+      validateOnMount: true,
+      onSubmit: async (values) => {
+        let res = await createWallet({
+          variables: {
+            input: {
+              customerId,
+              ...values,
+            },
+          },
+          refetchQueries: ['getCustomer'],
+        })
+
+        const { errors } = res
+
+        if (!errors) {
+          ;(ref as unknown as RefObject<DialogRef>)?.current?.closeDialog()
+        }
+      },
+    })
+
+    return (
+      <Dialog
+        ref={ref}
+        title={translate('text_62d18855b22699e5cf55f871')}
+        description={translate('text_62d18855b22699e5cf55f873')}
+        actions={({ closeDialog }) => (
+          <>
+            <Button
+              variant="quaternary"
+              onClick={() => {
+                closeDialog()
+                formikProps.resetForm()
+              }}
+            >
+              {translate('text_62d18855b22699e5cf55f89d')}
+            </Button>
+            <Button disabled={!formikProps.isValid} onClick={formikProps.submitForm}>
+              {translate('text_62d18855b22699e5cf55f89f')}
+            </Button>
+          </>
+        )}
+      >
+        <Content>
+          <TextInputField
+            name="name"
+            label={translate('text_62d18855b22699e5cf55f875')}
+            placeholder={translate('text_62d18855b22699e5cf55f877')}
+            formikProps={formikProps}
+          />
+          <InlineFields>
+            <TextInput value="1" label={translate('text_62d18855b22699e5cf55f879')} />
+            <TextInput value="=" />
+            <TextInputField
```

Comment:
This one should be disabled according to the design no ?

Reply:
Yes good catch ! done